tiatoolbox/dataloader/wsimeta.py

```diff
@@ -59,7 +59,7 @@ def __init__(
         self.vendor = vendor
         self.mpp = np.array(mpp)
         self.file_path = file_path
-        self.raw = dict(raw) if raw is not None else None
```

Comment:
Just a thought for the future, we should probably provide getter functions (or @properties) for these to stop a user accidentally modifying the original as they are returned by reference.

Reply:
Can we make it private or read-only?

Reply:
You cannot have private or read only variables in python. A leading underscore indicate private but does not prevent modification. A double preceding underscore performs 'name mangling'. See https://docs.python.org/3/tutorial/classes.html#private-variables

However, you can make a function with returns a copy and appears as if it is an attribute e.g.:

```python
@property
def raw(self):
    return copy.deepcopy(self.raw)

raw = wsi.raw # Gets a copy
raw["key"] = None # Modifies the copy but original is untouched
```

You can also use this method to memoize costly things e.g. getting slide info here (not particularly costly but an example):

```python
@property
def info(self):
    if not hasattr(self, "_m_info"):
        self._m_info = self._info()
    return self._m_info # or return a copy of to prevent modification
```